common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java

```diff
@@ -182,11 +182,13 @@ protected HttpResponse performTokenRequest(final GenericTokenRequest request) th
         if (request instanceof MicrosoftTokenRequest &&
                 !TextUtils.isEmpty(((MicrosoftTokenRequest) request).getBrokerVersion())) {
             headers.put(
-                    Device.PlatformIdParameters.BROKER_VERSION,
+                    AuthenticationConstants.PlatformIdParameters.BROKER_VERSION,
                     ((MicrosoftTokenRequest) request).getBrokerVersion()
             );
         }
-        headers.putAll(Device.getPlatformIdParameters());
+        headers.putAll(DiagnosticContext.getPlatformIdParameters());
+        headers.put(AuthenticationConstants.SdkPlatformFields.PRODUCT, DiagnosticContext.getRequestContext().get(AuthenticationConstants.SdkPlatformFields.PRODUCT));
```

Comment:
`headers.put(AuthenticationConstants.SdkPlatformFields.PRODUCT, DiagnosticContext.getRequestContext().get(AuthenticationConstants.SdkPlatformFields.PRODUCT));`

`headers.put(AuthenticationConstants.SdkPlatformFields.VERSION,DiagnosticContext.getRequestContext().get(AuthenticationConstants.SdkPlatformFields.VERSION));`

these can be part of `DiagnosticContext.getPlatformIdParameters()` as well right , why adding it seperately